src/pfe/file-watcher/server/src/projects/projectUtil.ts

```diff
@@ -400,6 +400,13 @@ async function executeBuildScript(operation: Operation, script: string, args: Ar
                     projectInfo.status = "success";
 
                     logger.logProjectInfo("The project location for " + operation.projectInfo.projectID + " is " + projectLocation, projectID, projectName);
+
+                    if (operation.projectInfo.projectType == "odo") {
```

Comment:
In all these if conditions, can you check if projectInfo.odoAppName is undefined, because we dont want to always set. We only want to do that, if odoAppName is not present

Reply:
Why do we not want to set it even if it is present? It will just override the old one which is the same?

Reply:
because once we have the name, is there a likelihood it will change? Otherwise we're doing an odo command execution everytime. The project handler getAppName calls an odo command to get the app name..

Reply:
hmm, that makes sense. i will add that